pkg/cloudprovider/provider/azure/create_delete_resources.go

```diff
@@ -0,0 +1,282 @@
+			}
+		}
+	}
+
+	return nil
+}
+
+func deleteDisksByMachineUID(ctx context.Context, c *config, machineUID types.UID) error {
+	disksClient, err := getDisksClient(c)
+	if err != nil {
+		return err
+	}
+
+	list, err := disksClient.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	allDisks := make([]compute.Disk, 0)
+
+	for {
+		allDisks = append(allDisks, list.Values()...)
+		if list.NotDone() {
+			list.Next()
+		} else {
+			break
+		}
+	}
+
+	for _, disk := range allDisks {
+		if disk.Tags != nil && *disk.Tags[machineUIDTag] == string(machineUID) {
+			future, err := disksClient.Delete(ctx, c.ResourceGroup, *disk.Name)
+			if err != nil {
+				return err
+			}
+
+			if err = future.WaitForCompletion(ctx, disksClient.Client); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func createPublicIPAddress(ctx context.Context, ipName string, machineUID types.UID, c *config) (network.PublicIPAddress, error) {
+	ipClient, err := getIPClient(c)
+	if err != nil {
+		return network.PublicIPAddress{}, err
+	}
+
+	ipParams := network.PublicIPAddress{
+		Name:     to.StringPtr(ipName),
+		Location: to.StringPtr(c.Location),
+		PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
+			PublicIPAddressVersion:   network.IPv4,
+			PublicIPAllocationMethod: network.Static,
+		},
+		Tags: map[string]*string{machineUIDTag: to.StringPtr(string(machineUID))},
+	}
+	future, err := ipClient.CreateOrUpdate(ctx, c.ResourceGroup, ipName, ipParams)
+	if err != nil {
+		return network.PublicIPAddress{}, fmt.Errorf("failed to create public IP address: %v", err)
+	}
+
+	err = future.WaitForCompletion(ctx, ipClient.Client)
+	if err != nil {
+		return network.PublicIPAddress{}, fmt.Errorf("failed to retrieve public IP address creation result: %v", err)
+	}
+
+	return future.Result(ipClient)
+}
+
+func getPublicIPAddress(ctx context.Context, ipName string, c *config) (network.PublicIPAddress, error) {
+	ipClient, err := getIPClient(c)
+	if err != nil {
+		return network.PublicIPAddress{}, err
+	}
+
+	return ipClient.Get(ctx, c.ResourceGroup, ipName, "")
+}
+
+func getSubnet(ctx context.Context, c *config) (network.Subnet, error) {
+	subnetsClient, err := getSubnetsClient(c)
+	if err != nil {
+		return network.Subnet{}, fmt.Errorf("failed to create subnets client: %v", err)
+	}
+
+	return subnetsClient.Get(ctx, c.ResourceGroup, c.VNetName, c.SubnetName, "")
+}
+
+func createNetworkInterface(ctx context.Context, ifName string, machineUID types.UID, config *config) (network.Interface, error) {
+	ifClient, err := getInterfacesClient(config)
+	if err != nil {
+		return network.Interface{}, fmt.Errorf("failed to create interfaces client: %v", err)
+	}
+
+	subnet, err := getSubnet(ctx, config)
+	if err != nil {
+		return network.Interface{}, fmt.Errorf("failed to fetch subnet: %v", err)
+	}
+
+	glog.Infof("Creating public IP for interface %q", ifName)
+	ipName := ifName + "-pubip"
+	_, err = createPublicIPAddress(ctx, ipName, machineUID, config)
```

Comment:
Do instances need a publicIP to be able to communicate with the Internet? I am wondering, because in e.G. openstack they do not, thus its optional to assign a public IP to the nodes. If you e.G. have your whole cluster inside Azure, there is most likely no reason for the instances to have a publicIP

Reply:
If it's not necessary - we should add a flag to optionally add a public ip

Reply:
For now I need it for debugging, but yes, I will create a flag.

Reply:
done